sci-physics/yoda/yoda-1.9.9.ebuild

```diff
@@ -0,0 +1,88 @@
+# Copyright 2022-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit bash-completion-r1 autotools python-single-r1 optfeature
+
+DESCRIPTION="Yet more Objects for (High Energy Physics) Data Analysis"
+HOMEPAGE="https://yoda.hepforge.org/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.com/hepcedar/yoda"
+else
+	SRC_URI="https://yoda.hepforge.org/downloads?f=${P^^}.tar.bz2 -> ${P^^}.tar.bz2"
+	S="${WORKDIR}/${P^^}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-3"
+SLOT="0/${PV}"
+IUSE="root test +python +zlib"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} ) root? ( python )"
+
+RDEPEND="
+	root? ( sci-physics/root:=[${PYTHON_SINGLE_USEDEP}] )
+	python? ( ${PYTHON_DEPS} )
+	zlib? ( sys-libs/zlib )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	python? (
+		$(python_gen_cond_dep '
+			>=dev-python/cython-0.29.24[${PYTHON_USEDEP}]
+		')
+		test? (
+			$(python_gen_cond_dep '
+				dev-python/numpy[${PYTHON_USEDEP}]
+				dev-python/matplotlib[${PYTHON_USEDEP}]
+			')
+		)
+	)
+"
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+	# reconf due to python3_10 patch
+	eautoreconf
+}
+
+src_configure() {
+	# we need to use the prefix cython here
+	econf --disable-static \
+		$(use_enable root) \
+		$(use_enable python pyext) \
+		$(use_with zlib zlib "${ESYSROOT}/usr") \
+		$(usex python CYTHON="${ESYSROOT}/usr/bin/cython")
+}
+
+src_test() {
+	# PYTESTS and SHTESTS both require python tools
+	if use python; then
+		emake check
+	else
+		emake check PYTESTS= SHTESTS= NO_PYTHON=1
+	fi
+}
+
+src_install() {
+	emake install DESTDIR="${D}"
+
+	newbashcomp "${ED}"/etc/bash_completion.d/${PN}-completion ${PN}
```

Comment:
We have this bug open about the bash-completions? Could you look into fixing it for the new versions? 

https://bugs.gentoo.org/926510

Reply:
[test_sci-physics_yoda-1.9.9_amd64.log](https://github.com/gentoo/gentoo/files/14616942/test_sci-physics_yoda-1.9.9_amd64.1.log)
[test_sci-physics_yoda-1.9.10_amd64.log](https://github.com/gentoo/gentoo/files/14616948/test_sci-physics_yoda-1.9.10_amd64.log)

I only get, so I guess upstream fixed bash completions:
```
 * QA Notice: setuptools warnings detected:
 * 
 * 	Package 'cppy.include.cppy' is absent from the `packages` configuration.
 ```

Reply:
Also all my tests passed there, I don't know why they fail for you...

Reply:
> Also (as you can see) all my tests passed there, I don't know why they fail for you...

Because yoda is not yet installed on the system. If I would install yoda first with `USE=-test`, then run it again with the tests then it would be able to find the yoda python module.

Reply:
But the first line of the log is 
````
[ebuild  N    ~] sci-physics/yoda-1.9.10:0/1.9.10::science  USE="test zlib -python -root" PYTHON_SINGLE_TARGET="python3_10 python3_11 -python3_12" 1249 KiB
```
No `U` no `D` no `R`. I ran it in a gitlab CI with a fresh image w/o yoda.

Reply:
Yes but this first one is built with `USE=-python`  so these tests aren't run.

The later one that does run the python tests is a rebuild (`R`):
```
[ebuild   R   ~] sci-physics/yoda-1.9.10:0/1.9.10::science  USE="python* test -root -zlib*" PYTHON_SINGLE_TARGET="python3_12 -python3_10* -python3_11" 0 KiB
```

Reply:
I see pkg-testing-tools first runs w/o FEATURES="test" and after that with it.

Reply:
Aaah that would explain it. 

I think that behaviour is wrong though. Sometimes the tests depend on things that the package installs, in that case we often need to set some variable to have the tests run successfully on "clean" installs. `PYTHONPATH` is a common one, but also sometimes `PATH` for tests that call binaries installed by the package.